### PR TITLE
refactor(Core/Misc): Define & Rename Attributes and flags

### PR DESCRIPTION
**Pull request by Kitzunu**
Co-Authored-By: offl <11556157+offl@users.noreply.github.com>
Co-Authored-By: Treeston <treeston.mmoc@gmail.com>

Cherry-Pick https://github.com/TrinityCore/TrinityCore/commit/e6d9d9ead99aba4bfbd0b6389cb7a88827fe40d7
Cherry-Pick https://github.com/TrinityCore/TrinityCore/commit/081eab3cf5a5ec42917b2a7eaebf816fb36da423

## Changes Proposed:
-  Rename and define some attributes and flags.
- Blizzlike names based on leaked enum db2 file from the tbc classic client.

## Tests Performed:

- None

## How to Test the Changes:

Not necessary. Just name changes

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Miscellaneous/SharedDefines.h [hunk 1]
```diff
@@ -2571,38 +2571,38 @@ enum CreatureFamily
...
+    CREATURE_TYPE_FLAG_IS_LINK_ALL                       = 0x00400000,
+    CREATURE_TYPE_FLAG_INTERACT_ONLY_WITH_CREATOR        = 0x00800000,
+    CREATURE_TYPE_FLAG_DO_NOT_PLAY_UNIT_EVENT_SOUNDS     = 0x01000000,
+    CREATURE_TYPE_FLAG_HAS_NO_SHADOW_BLOB                = 0x02000000,
+    CREATURE_TYPE_FLAG_TREAT_AS_RAID_UNIT                = 0x04000000,   // ! Creature can be targeted by spells that require target to be in caster's party/raid
+    CREATURE_TYPE_FLAG_FORCE_GOSSIP                      = 0x08000000,   // Allows the creature to display a single gossip option.
+    CREATURE_TYPE_FLAG_DO_NOT_SHEATHE                    = 0x10000000,
+    CREATURE_TYPE_FLAG_DO_NOT_TARGET_ON_INTERACTION      = 0x20000000,
+    CREATURE_TYPE_FLAG_DO_NOT_RENDER_OBJECT_NAME         = 0x40000000,
+    CREATURE_TYPE_FLAG_UNIT_IS_QUEST_BOSS                = 0x80000000    // Not verified
```
Comment (FrancescoBorzi) [inline src/server/game/Miscellaneous/SharedDefines.h:2605]: @Kitzunu have you double-checked that all of these renames have really the same meaning? also what about labels in Keira3 ?

Comment (FrancescoBorzi) [reply src/server/game/Miscellaneous/SharedDefines.h:2605]: > the devil itself when it comes to changing things...

says the guy who never read the documentation :D

but seriously... to change those names it's very easy, look at: https://github.com/azerothcore/Keira3/blob/master/src/app/shared/constants/flags/creature-type-flags.ts

Comment (Kitzunu) [reply src/server/game/Miscellaneous/SharedDefines.h:2605]: I know, I use Notepad++ and rename everything that matches :D

fast simple and always correct

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/Miscellaneous/SharedDefines.h (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/game/Miscellaneous/SharedDefines.h:2605]: @Kitzunu please prepare the Keira3 PR for this rename and this is ready to go

Comment (Kitzunu) [reply src/server/game/Miscellaneous/SharedDefines.h:2605]: https://github.com/azerothcore/Keira3/pull/1160 @FrancescoBorzi

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**